### ISSUE TITLE
points_in_polygon

### ISSUE THREAD

**Issue by lfriedri** [enthought/chaco#351]
in chaco 4.6.1 points_in_polygon is imported from kiva.agg (in polygon_plot.py and in tools/lasso_selection.py). This import can not be resolved in enable 4.6.2 since points_in_polygon was moved to kiva.api.

**Comment by corranwebster** [comment]
Thanks for the report.  The fix should be straightforward, but it may take us a little while to get out a bugfix release for Chaco.

**Comment by corranwebster** [comment]
@lfriedri Can you please double-check that you are in fact using the released version of Enable 4.6.2 and not the current master branch.  I have downloaded the tarball from the release of Enable 4.6.2 and there is no `kiva.api` in the tarball (as expected) and `points_in_polygon` appears to be in `kiva.agg` as expected.

**Comment by corranwebster** [comment]
OK, I think I understand the confusion (please correct me if I am wrong) - the current master branch of Enable incorrectly claims to be 4.6.2.dev.  We will fix this so that master of Enable reports 4.7.0.dev.

Also there is of course the original import problems which will need to be fixed, but won't require a bugfix release of Chaco.

Apologies for any confusion.

**Comment by lfriedri** [comment]
I double-checked:
enable.__version__ gives 4.6.2
To install it, I downloaded enable-4.6.2-cp36-cp36m-win_amd64.whl from http://www.lfd.uci.edu/~gohlke/pythonlibs/

**Comment by lfriedri** [comment]
When I change the imports manually to "from kiva.api import points_in_polygon" I get the following traceback when running simple_line.py from chaco/examples:

`Microsoft Windows [Version 10.0.10586]
(c) 2015 Microsoft Corporation. All rights reserved.

C:\Users\my_name>cd C:\Users\my_name\Documents\my_project

C:\Users\my_name\Documents\my_project>simple_line.bat

C:\Users\my_name\Documents\my_project>"WinPython-64bit-3.6.1.0Zero\scripts\python.bat" "simple_line.py"

C:\Users\my_name\Documents\my_project>simple_line.bat

C:\Users\my_name\Documents\my_project>"WinPython-64bit-3.6.1.0Zero\scripts\python.bat" "simple_line.py"
Traceback (most recent call last):
  File "simple_line.py", line 150, in <module>
    demo.configure_traits()
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traits\has_traits.py", line 2171, in configure_traits
    kind, handler, id, scrollable, args )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\toolkit.py", line 213, in view_application
    id, scrollable, args )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\view_application.py", line 83, in view_application
    scrollable, args ).ui.result
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\view_application.py", line 134, in __init__
    args       = self.args )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\view.py", line 433, in ui
    ui.ui( parent, kind )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\ui.py", line 222, in ui
    self.rebuild( self, parent )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\toolkit.py", line 155, in ui_live
    ui_live.ui_live( ui, parent )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_live.py", line 39, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_live.py", line 59, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_base.py", line 482, in display_ui
    ui.owner.init(ui, parent, style)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_live.py", line 177, in init
    self.add_contents(panel(ui), bbox)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 255, in panel
    panel = _GroupPanel(content[0], ui).control
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 592, in __init__
    layout = self._add_items(content, inner)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\qt4\ui_panel.py", line 852, in _add_items
    ui, object, name, item.tooltip, None
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\editor_factory.py", line 145, in simple_editor
    description = description )
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traitsui\editor.py", line 133, in __init__
    self.old_value = getattr( self.object, self.name )
  File "simple_line.py", line 143, in _plot_default
    bgcolor="lightgray", use_backbuffer=True)
  File "simple_line.py", line 45, in __init__
    self._setup_plots()
  File "simple_line.py", line 61, in _setup_plots
    color=tuple(COLOR_PALETTE[i]))
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\chaco\plot_factory.py", line 58, in create_scatter_plot
    index, value = _create_data_sources(data)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\chaco\plot_factory.py", line 29, in _create_data_sources
    index = ArrayDataSource(array(index), sort_order=index_sort)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\chaco\array_data_source.py", line 110, in __init__
    self.set_data(data, sort_order)
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\chaco\array_data_source.py", line 123, in set_data
    self._data = newdata
  File "C:\Users\my_name\Documents\my_project\WinPython-64bit-3.6.1.0Zero\python-3.6.1.amd64\lib\site-packages\traits\trait_handlers.py", line 172, in error
    value )
traits.trait_errors.TraitError: The '_data' trait of an ArrayDataSource instance must be an array or None, but a value of array([ -5.00000000e+00,  -4.80000000e+00,  -4.60000000e+00,
        -4.40000000e+00,  -4.20000000e+00,  -4.00000000e+00,
        -3.80000000e+00,  -3.60000000e+00,  -3.40000000e+00,
        -3.20000000e+00,  -3.00000000e+00,  -2.80000000e+00,
        -2.60000000e+00,  -2.40000000e+00,  -2.20000000e+00,
        -2.00000000e+00,  -1.80000000e+00,  -1.60000000e+00,
        -1.40000000e+00,  -1.20000000e+00,  -1.00000000e+00,
        -8.00000000e-01,  -6.00000000e-01,  -4.00000000e-01,
        -2.00000000e-01,   4.44089210e-15,   2.00000000e-01,
         4.00000000e-01,   6.00000000e-01,   8.00000000e-01,
         1.00000000e+00,   1.20000000e+00,   1.40000000e+00,
         1.60000000e+00,   1.80000000e+00,   2.00000000e+00,
         2.20000000e+00,   2.40000000e+00,   2.60000000e+00,
         2.80000000e+00,   3.00000000e+00,   3.20000000e+00,
         3.40000000e+00,   3.60000000e+00,   3.80000000e+00,
         4.00000000e+00,   4.20000000e+00,   4.40000000e+00,
         4.60000000e+00,   4.80000000e+00,   5.00000000e+00,
         5.20000000e+00,   5.40000000e+00,   5.60000000e+00,
         5.80000000e+00,   6.00000000e+00,   6.20000000e+00,
         6.40000000e+00,   6.60000000e+00,   6.80000000e+00,
         7.00000000e+00,   7.20000000e+00,   7.40000000e+00,
         7.60000000e+00,   7.80000000e+00,   8.00000000e+00,
         8.20000000e+00,   8.40000000e+00,   8.60000000e+00,
         8.80000000e+00,   9.00000000e+00,   9.20000000e+00,
         9.40000000e+00,   9.60000000e+00,   9.80000000e+00,
         1.00000000e+01,   1.02000000e+01,   1.04000000e+01,
         1.06000000e+01,   1.08000000e+01,   1.10000000e+01,
         1.12000000e+01,   1.14000000e+01,   1.16000000e+01,
         1.18000000e+01,   1.20000000e+01,   1.22000000e+01,
         1.24000000e+01,   1.26000000e+01,   1.28000000e+01,
         1.30000000e+01,   1.32000000e+01,   1.34000000e+01,
         1.36000000e+01,   1.38000000e+01,   1.40000000e+01,
         1.42000000e+01,   1.44000000e+01,   1.46000000e+01,
         1.48000000e+01,   1.50000000e+01]) <class 'numpy.ndarray'> was specified.`

**Comment by corranwebster** [comment]
I just downloaded the wheel from http://www.lfd.uci.edu/~gohlke/pythonlibs/ and it does not match the Enable 4.6.2 tar.gz on PyPI at https://pypi.python.org/pypi/enable/4.6.2.  For whatever reason, they are not providing a correct build for the released 4.6.2 version of Enable.  You may wish to contact the maintainer of that website to let them know of the problem.

If you have the appropriate toolchain installed (Microsoft's Python compiler tools and SWIG being the main ones) you should be able to get a correct version of Enable 4.6.2 via `pip install`.

**Comment by lfriedri** [comment]
Thank you for chasing this down.
I agree that https://pypi.python.org/pypi/enable/4.6.2 does not have kiva.api while the version from  http://www.lfd.uci.edu/~gohlke/pythonlibs/ has it. I will ask them if they made a mistake.

Just for my understanding: If I go to https://github.com/enthought/enable/tree/master/kiva, I actually DO find a api.py that imports points_in_polygon. But this version was not released yet, correct?

**Comment by corranwebster** [comment]
Yes, your understanding is correct.  The PR which is responsible for the change is this one: https://github.com/enthought/enable/pull/250 which was merged last week, but was not in the 4.6.2 bugfix release we put out on Tuesday.

We will be fixing Chaco's master in the near future to bring it into line with the new Enable API, so your bug report was valuable in letting us know that there were problems between the two master branches that we needed to address.

Thank you for bringing this to our attention!
